File: Next-Greater-Element-I.java
1class Solution {
2    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
3        int n1 = nums1.length, n2 = nums2.length;
4        List<Integer> arr = new ArrayList<>();
5        for(int idx1 = 0 ; idx1 < n1 ; idx1++){
6            boolean flag = false;
7            for(int idx2 = 0 ; idx2 < n2 ; idx2++){
8                if(nums1[idx1] == nums2[idx2]){
9                    int k = idx2;
10                    while(k < n2){
11                        if(nums2[k] > nums2[idx2]){
12                            arr.add(nums2[k]);
13                            flag = true;
14                            break;
15                        }
16                        k++;
17                    }
18                    if(!flag) arr.add(-1);
19                }
20            }
21        }
22        int[] ans = new int[n1];
23        for(int index = 0 ; index < n1 ; index++) ans[index] = arr.get(index);
24        return ans;
25    }
26}

File: Next-Greater-Element-I.cpp
1class Solution {
2public:
3    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
4        vector<int> ans;
5        unordered_map<int,int> nge;
6        stack<int> st;
7        int n = nums2.size();
8        for(int index = n - 1;  index >= 0 ; index--){
9            while(!st.empty() && st.top() < nums2[index]) st.pop();
10            if(st.empty()) nge[nums2[index]] = -1;
11            else{
12                nge[nums2[index]] = st.top();
13            }
14            st.push(nums2[index]);
15        }
16        for(int num : nums1) ans.push_back(nge[num]);
17        return ans;
18    }
19};